### PR TITLE
[Form] Changed UrlType input type to text when default_protocol is not null

### PR DESCRIPTION
**Pull request by MatTheCat**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #29690 
| License       | MIT

### CODE REVIEW

**Comment by xabbuh** [comment]
@MatTheCat it looks like some tests need to be fixed

Status: Needs work

**Comment by MatTheCat** [comment]
@xabbuh I don't understand what's going on. I see tests failing on `Symfony\Bridge\Twig\Tests\Extension\FormExtensionBootstrap3HorizontalLayoutTest::testUrl` but this test doesn't exist anymore in this PR. Why do I see it?

**Comment by ro0NL** [comment]
we might gain some extra value to use `inputmode=url` when using `type=text`. AFAIK this will force a specific keyboard on mobile (e.g. URL input).

https://caniuse.com/#feat=input-inputmode

**Comment by ro0NL** [comment]
tested on android with built-in samsung keyboard, and works as expected :+1:

**Comment by MatTheCat** [comment]
Don't know if failing tests prevent to merge this?

**Review by xabbuh — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Bridge/Twig/Tests/Extension/FormExtensionTableLayoutTest.php [hunk 1]
```diff
@@ -76,6 +76,34 @@ public function testStartTagHasActionAttributeWhenActionIsZero()
         $this->assertSame('<form name="form" method="get" action="0">', $html);
     }
 
+    public function testUrlWithDefaultProtocol()
```
Comment (stof) [inline src/Symfony/Bridge/Twig/Tests/Extension/FormExtensionTableLayoutTest.php]: I don't think you should duplicate them in all components. Put them in AbstractLayoutTest instead, where the old test was.

**Review by stof — CHANGES_REQUESTED** [review]

File: src/Symfony/Bridge/Twig/Tests/Extension/FormExtensionTableLayoutTest.php (hunk 1 above)
Comment (MatTheCat) [reply src/Symfony/Bridge/Twig/Tests/Extension/FormExtensionTableLayoutTest.php]: I guess @xabbuh had a reason?

Comment (stof) [reply src/Symfony/Bridge/Twig/Tests/Extension/FormExtensionTableLayoutTest.php]: Well, the commit doing that says "fix deps=high". But I think that having them in AbstractLayoutTest will work fine for deps=high once the updated code is merged up to the highest version too.

Comment (MatTheCat) [reply src/Symfony/Bridge/Twig/Tests/Extension/FormExtensionTableLayoutTest.php]: Reverting then.

Comment (xabbuh) [reply src/Symfony/Bridge/Twig/Tests/Extension/FormExtensionTableLayoutTest.php]: @MatTheCat I rebased and removed the commit entirely instead.

**Comment by MatTheCat** [comment]
@stof is it good now?

**Review by ro0NL — APPROVED** [review]

**Comment by MatTheCat** [comment]
@stof ?

**Comment by MatTheCat** [comment]
@fabpot could this PR be merged please?

**Comment by mablae** [comment]
@MatTheCat If you need to use your branch NOW, you could alias it via composer.json. 

Pushing for merge and mentioning fabpot seldom results in a faster merge. If the feature is good it will be merged. Please be a little more patient after putting lot of work into your PR. 

And thank you for the work!

**Comment by MatTheCat** [comment]
I conclude the feature is not good so there's no point for me to continue rebasing my branch.

**Comment by mablae** [comment]
You're wrong. You already got positive feedback.

**Comment by MatTheCat** [comment]
Dude don't provoke me on my personal email. You're point is a PR is merged if it's good right? This one didn't.

Can we stop the conversation here?

**Pull request closed without merging**